src/components/organisms/LocalReplace.tsx

```diff
@@ -0,0 +1,438 @@
+      onReplaceQueryChange(replaceValue)
+    }
+  }, [onReplaceQueryChange, replaceValue])
+
+  const focusReplaceTextAreaInput = useCallback(
+    (replaceValueLength?: number) => {
+      if (replaceTextAreaRef.current == null) {
+        return
+      }
+      replaceTextAreaRef.current.focus()
+      if (replaceValueLength && replaceValueLength > 0) {
+        replaceTextAreaRef.current.selectionEnd += replaceValueLength
+        replaceTextAreaRef.current.selectionStart = replaceValueLength
+      }
+    },
+    []
+  )
+
+  const replaceSingleMarker = useCallback(
+    (marker: TextMarker) => {
+      const markerPos: MarkerRange = marker.find() as MarkerRange
+      if (!markerPos) {
+        return false
+      }
+      const foundMarkText = codeMirror.getRange(markerPos.from, markerPos.to)
+      if (foundMarkText.length >= 1 && preserveCaseReplace) {
+        // if first character was upper case - set the replacement to upper case too
+        codeMirror.replaceRange(
+          foundMarkText.charAt(0) + replaceQuery.substr(1),
+          markerPos.from,
+          markerPos.to,
+          '@ignore'
+        )
+      } else {
+        codeMirror.replaceRange(
+          replaceQuery,
+          markerPos.from,
+          markerPos.to,
+          '@ignore'
+        )
+      }
+
+      return true
+    },
+    [codeMirror, preserveCaseReplace, replaceQuery]
+  )
+
+  const findSelectedMarker = useCallback(() => {
+    for (const marker of codeMirror.getAllMarks()) {
+      if (marker.className === 'marked selected') {
+        return marker
+      }
+    }
+    return null
+  }, [codeMirror])
+
+  const onReplaceCurrentItem = useCallback(() => {
+    const marker = findSelectedMarker()
+    if (!marker) {
+      return
+    }
+
+    const replaceSuccessful = replaceSingleMarker(marker)
+    if (onReplacementFinished && replaceSuccessful) {
+      onReplacementFinished()
+    }
+    focusReplaceTextAreaInput(replaceQuery.length)
+  }, [
+    findSelectedMarker,
+    focusReplaceTextAreaInput,
+    onReplacementFinished,
+    replaceQuery.length,
+    replaceSingleMarker,
+  ])
+
+  const onReplaceAll = useCallback(() => {
+    const markers = codeMirror.getAllMarks()
+    if (markers.length <= 0) {
+      return
+    }
+
+    let anyReplacementSuccessful = false
+    for (const marker of markers) {
+      const replaceSuccessful = replaceSingleMarker(marker)
+      if (replaceSuccessful) {
+        anyReplacementSuccessful = true
+      }
+    }
+
+    if (onReplacementFinished && anyReplacementSuccessful) {
+      onReplacementFinished()
+    }
+  }, [codeMirror, onReplacementFinished, replaceSingleMarker])
+
+  const handleExcludeCurrentItem = useCallback(() => {
+    navigateToNext(SearchResultNavigationDirection.NEXT)
+    focusReplaceTextAreaInput(replaceQuery.length)
+  }, [focusReplaceTextAreaInput, navigateToNext, replaceQuery.length])
+
+  const toggleCaseSensitiveReplace = useCallback(
+    (focusOnInput = true) => {
```

Comment:
I had a lot of problems with the focus implementation, so for now we don't need it. 
I'll remove it and if it will be needed it can be easily added in the future.